### PR TITLE
[Compute] `az vmss create/update`: Add new parameters `--enable-resilient-creation` and `--enable-resilient-deletion` to support Resiliency Policy on VMSS

### LINKED ISSUE #28451: CLI Support for ResiliencyPolicy (including ResilientVMCreationPolicy, ResilientVMDeletionPolicy) on VMSS
### Preconditions

- [X] No need to upgrade Python SDK or the Python SDK is ready.

### Related command

_No response_

### Resource Provider

Microsoft.Compute/ComputeRP

### Description of Feature or Work Requested

Our ask is to have CLI support for ResiliencyPolicy (including ResilientVMCreationPolicy, ResilientVMDeletionPolicy) which is a new feature on VMSS.

Resilient VM Create will automatically recover customers from OS Provisioning Timeout and VM Start Timeout errors experienced during a VM Create operation by deleting and recreating the affected VM. Resilient VM Delete will retry VM Delete requests asynchronously in the event of a failed delete operation. This feature will be available on VMSS Uniform and Flex.

API Design:
```
{ 
    "properties": { 
        "ResiliencyPolicy": { 
          "ResilientVMCreationPolicy": {
                 "Enabled": true 
           }, 
           "ResilientVMDeletionPolicy":{
              "Enabled": true
            } 
       }
    }
}
```

If Enabled == true for ResilientVMCreationPolicy, VMSS will automatically delete + recreate when an instance goes into OS Provisioning Timeout or VM Start Timeout during the creation process. VMSS will retry the create operation for a set retryLimit (configured into the backend, not exposed to customers). If Enabled == false, the instance will fall into "Provisioning Failed" state if OS Provisioning Timeout or VM Start Timeout is encountered during the VM create process.

If Enabled == true for ResilientVMDeletionPolicy, VMSS will automatically retry delete operations asynchronously if the initial VM deletion fails. If Enabled == false, the VM will fall into a "Failed"/unusable state if delete operation fails.

Dev Spec, Resilient VM Create: https://microsoft.sharepoint.com/:w:/t/ComputeVM/EWXZuEBUcm9Gvl6e7e0PlcQBAMwkAODWW1b5Z5xDdf1GuA?e=Qy8Vzb

Dev Spec, Resilient VM Delete: [Reliable VM Deletion.docx](https://microsoft.sharepoint.com/:w:/t/ComputeVM/EVzT8JkP-3ZAgwy-jmpSJnsBP_DY7YWCLFoBTCE-4jNUSQ?e=sPbne6)

PM Spec: https://microsoft.sharepoint.com/:w:/t/AzureComputeInfrastructurePlatformTeam/EZNWFEW6xONLiWMFq33asfoB_FfF0YwjVbuBg-xCQwzbXA?e=44GuXz

### Minimum API Version Required

2023-09-01

### Swagger PR link / SDK link

https://github.com/Azure/azure-rest-api-specs/blob/c75671ae0a5d82d7d96fda75150478e8581408d8/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/virtualMachineScaleSet.json#L3914](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fblob%2Fc75671ae0a5d82d7d96fda75150478e8581408d8%2Fspecification%2Fcompute%2Fresource-manager%2FMicrosoft.Compute%2FComputeRP%2Fstable%2F2023-09-01%2FvirtualMachineScaleSet.json%23L3914&data=05%7C02%7Chilarywang%40microsoft.com%7Cd317c22e543441a7c45008dc34d195bd%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638443321975559884%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=g5FAWylQ%2FZJyk60bQfzf%2FT4w9GPwXaHiBwCSMY5X%2F0s%3D&reserved=0

### Request Example

_No response_

### Target Date

2024-06-21

### PM Contact

hilarywang

### Engineer Contact

raredd

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ReaNAiveD**
**Related command**

* `vmss create`
* `vmss update`

**Description**

Resolve #28451 
> Our ask is to have CLI support for ResiliencyPolicy (including ResilientVMCreationPolicy, ResilientVMDeletionPolicy) which is a new feature on VMSS.
> Resilient VM Create will automatically recover customers from OS Provisioning Timeout and VM Start Timeout errors experienced during a VM Create operation by deleting and recreating the affected VM. Resilient VM Delete will retry VM Delete requests asynchronously in the event of a failed delete operation. This feature will be available on VMSS Uniform and Flex.

**Testing Guide**

```bash
az vmss create --debug -g myRG -n myVMSS --image MicrosoftWindowsServer:WindowsServer:2016-Datacenter:latest --vm-sku Standard_D1_v2 --admin-username vmtest --admin-password Test1Password# --enable-resilient-vm-creation --disable-overprovision --upgrade-policy-mode Manual --orchestration-mode Uniform
az vmss update -g myRG -n myVMSS --enable-resilient-vm-deletion --enable-resilient-vm-creation false
```

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -861,6 +861,8 @@ def load_arguments(self, _):
             c.argument('regular_priority_count', type=int, min_api='2022-08-01', is_preview=True, help='The base number of regular priority VMs that will be created in this scale set as it scales out. Must be greater than 0.')
             c.argument('regular_priority_percentage', type=int, min_api='2022-08-01', is_preview=True, help='The percentage of VM instances, after the base regular priority count has been reached, that are expected to use regular priority. Must be between 0 and 100.')
             c.argument('enable_osimage_notification', arg_type=get_three_state_flag(), min_api='2022-11-01', help='Specify whether the OS Image Scheduled event is enabled or disabled.')
+            c.argument('enable_resilient_vm_creation', arg_type=get_three_state_flag(), min_api='2023-09-01', help='Automatically recover customers from OS Provisioning Timeout and VM Start Timeout errors experienced during a VM Create operation by deleting and recreating the affected VM.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:864]:
```suggestion
            c.argument('enable_resilient_creation', arg_type=get_three_state_flag(), min_api='2023-09-01', help='Automatically recover customers from OS Provisioning Timeout and VM Start Timeout errors experienced during a VM Create operation by deleting and recreating the affected VM.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -861,6 +861,8 @@ def load_arguments(self, _):
...
+            c.argument('enable_resilient_vm_deletion', arg_type=get_three_state_flag(), min_api='2023-09-01', help='Retry VM Delete requests asynchronously in the event of a failed delete operation.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:865]:
```suggestion
            c.argument('enable_resilient_deletion', arg_type=get_three_state_flag(), min_api='2023-09-01', help='Retry VM Delete requests asynchronously in the event of a failed delete operation.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:864]: It would be better if we can make the parameter name more concise

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**